Analyzer/src/TwoLepAnalyzer.cc

```diff
@@ -0,0 +1,442 @@
+        const auto& deepESM_bin4        = tr.getVar<bool>("deepESM_bin4");
+        const auto& deepESM_binNum      = tr.getVar<int>("deepESM_binNum");
+        const auto& onZ                 = tr.getVar<bool>("onZ");
+
+        // ------------------------
+        // -- Define weight
+        // ------------------------
+        double weight               = 1.0;
+        double eventweight          = 1.0;
+        double leptonScaleFactor    = 1.0;
+        double bTagScaleFactor      = 1.0;
+        double htDerivedScaleFactor = 1.0;
+        double topPtScaleFactor     = 1.0;
+        double prefiringScaleFactor = 1.0;
+        double puScaleFactor        = 1.0;
+        
+        if(runtype == "MC")
+        {
+            // Define Lumi weight
+            const auto& Weight  = tr.getVar<double>("Weight");
+            const auto& lumi = tr.getVar<double>("Lumi");
+            eventweight = lumi*Weight;
+            
+            // Define lepton weight
+            if(NGoodLeptons == 1)
+            {
+                const auto& eleLepWeight = tr.getVar<double>("totGoodElectronSF");
+                const auto& muLepWeight  = tr.getVar<double>("totGoodMuonSF");
+                leptonScaleFactor = (GoodLeptons[0].first == "e") ? eleLepWeight : muLepWeight;
+            }
+            
+            //PileupWeight = tr.getVar<double>("_PUweightFactor");
+            bTagScaleFactor   = tr.getVar<double>("bTagSF_EventWeightSimple_Central");
+            htDerivedScaleFactor = tr.getVar<double>("htDerivedweight");
+            topPtScaleFactor = tr.getVar<double>("topPtScaleFactor");
+            prefiringScaleFactor = tr.getVar<double>("prefiringScaleFactor");
+            puScaleFactor = tr.getVar<double>("puWeightCorr");
+            
+            weight *= eventweight*leptonScaleFactor*bTagScaleFactor*htDerivedScaleFactor*prefiringScaleFactor*puScaleFactor;
+        }
+        // Create vector of only bjet TLorentzVectors
+        std::vector<TLorentzVector> GoodBJetsVec_pt30;
+        for (int j=0; j < Jets.size(); j++)
+        {
+            if (GoodBJets_pt30.at(j))
+            {
+                GoodBJetsVec_pt30.push_back(Jets.at(j));
+            }
+        }
+        // two lepton M(b,l) definition
+        TLorentzVector lep1;
+        TLorentzVector lep2;
+        int  used_jet;
+        double Mbl1;
+        double Mbl2;
+        double mbl1diff=999;
+        double mbl2diff=999;
+        if(NGoodLeptons==2)
+        {
+            if(GoodLeptons.at(0).second.Pt() >= GoodLeptons.at(1).second.Pt())
+            {
+                lep1 = GoodLeptons.at(0).second;
+                lep2 = GoodLeptons.at(1).second;
+            }
+            else
+            {
+                lep1 = GoodLeptons.at(1).second;
+                lep2 = GoodLeptons.at(0).second;
+            }
+            if(NGoodBJets_pt30 >= 2)
+            {
+                for(int b=0; b < NGoodBJets_pt30; b++)
```

Comment:
This works fine but you can use a range base for loop:
for(const auto& b : GoodBJetVec_pt30)
{
    double mbl1 = (lep1+b).M();
    .....
}

Reply:
I use an integer index so I know which bjet was used, since I can't compare the TLorentzVectors to make sure the bjet isn't reused.